From: Jan Kara <jack@suse.com>
Subject: dmapi: Fix xfs dmapi to not unlock & lock XFS_ILOCK_EXCL
References: bsc#949744
Patch-mainline: Never, dmapi never went upstream

Dmapi code in xfs_file_write_checks() releases XFS_ILOCK_EXCL and acquires
it again after sending dmapi event. However we no longer hold XFS_ILOCK_EXCL
in xfs_file_write_checks() so this is just wrong. Remove the locking dance.

Signed-off-by: Jan Kara <jack@suse.com>

---
 fs/xfs/xfs_file.c |    2 --
 1 file changed, 2 deletions(-)

--- a/fs/xfs/xfs_file.c
+++ b/fs/xfs/xfs_file.c
@@ -638,12 +638,10 @@ restart:
 		if (*iolock & XFS_IOLOCK_EXCL)
 			 dmflags |= DM_FLAGS_IMUX; /* dmapi disable mutex flg */
 
-		xfs_rw_iunlock(ip, XFS_ILOCK_EXCL); /* does not disable mutex */
 		error = XFS_SEND_DATA(ip->i_mount, DM_EVENT_WRITE, ip,
 				      *pos, *count, dmflags, iolock);
 		if (error)
 			return -error;
-		xfs_rw_ilock(ip, XFS_ILOCK_EXCL);
 
 	/* DMAPI NOSPACE will call this routine again. eventsent is retained
 	 * to ensure that DM event is only sent once.
